Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/RoslynTargetsTests.cs

```diff
@@ -42,45 +42,150 @@ public class RoslynTargetsTests
         #region SetRoslynSettingsTarget tests
 
         [TestMethod]
-        [Description("Checks the happy path i.e. settings exist in config, no reason to exclude the project")]
-        public void Roslyn_Settings_ValidSetup()
```

Comment:
I've replaced the old test with some new ones that check the combinations of test/non-test, C#/VB.